### PR TITLE
br: add use constraint with CDC and drainer

### PR DESCRIPTION
**Pull request by YuJuncen**
### What is changed, added or deleted? (Required)

现今 CDC 和 Drainer 无法同步 BR 所恢复的数据，同时 BR 恢复到上游集群可能会让 CDC / Drainer 工作异常。

于是在文档中标注出来。

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### CODE REVIEW

File: br/backup-and-restore-faq.md [hunk 1]
```diff
@@ -57,3 +57,11 @@ category: FAQ
 备份的时候仅仅在每个 Region 的 Leader 处生成该 Region 的备份文件。因此备份的大小等于数据大小，不会有多余的副本数据。所以最终的总大小大约是 TiKV 数据总量除以副本数。
  
 但是假如想要从本地恢复数据，因为每个 TiKV 都必须要能访问到所有备份文件，在最终恢复的时候会有等同于恢复时 TiKV 节点数量的副本。
+
+## BR 恢复到 CDC / Drainer 的上游集群时，要注意些什么？
```
Comment (TomShawn) [inline br/backup-and-restore-faq.md:61]:
```suggestion
## BR 恢复到 TiCDC / Drainer 的上游集群时，要注意些什么？
```

File: br/backup-and-restore-faq.md [hunk 2]
```diff
@@ -57,3 +57,11 @@ category: FAQ
...
+
+首先，**BR 恢复的数据无法被同步到下游**，因为 BR 直接导入 SST 文件，而下游集群目前没有办法获得上游的 SST 文件。
+
+另一方面，无法被同步到下游的恢复数据可能导致 CDC / Drainer 在执行 DDL 的时候发生异常。所以，如果一定要在 CDC / Drainer 的上游集群执行恢复，请将 BR 恢复的所有表加入 CDC / Drainer 的阻止名单。
```
Comment (TomShawn) [inline br/backup-and-restore-faq.md:65]:
```suggestion
+ 无法被同步到下游的恢复数据可能导致 CDC / Drainer 在执行 DDL 的时候发生异常。所以，如果一定要在 TiCDC / Drainer 的上游集群执行恢复，请将 BR 恢复的所有表加入 TiCDC / Drainer 的阻止名单。
```

File: br/backup-and-restore-faq.md [hunk 3]
```diff
@@ -57,3 +57,11 @@ category: FAQ
 备份的时候仅仅在每个 Region 的 Leader 处生成该 Region 的备份文件。因此备份的大小等于数据大小，不会有多余的副本数据。所以最终的总大小大约是 TiKV 数据总量除以副本数。
  
 但是假如想要从本地恢复数据，因为每个 TiKV 都必须要能访问到所有备份文件，在最终恢复的时候会有等同于恢复时 TiKV 节点数量的副本。
+
+## BR 恢复到 CDC / Drainer 的上游集群时，要注意些什么？
+
+首先，**BR 恢复的数据无法被同步到下游**，因为 BR 直接导入 SST 文件，而下游集群目前没有办法获得上游的 SST 文件。
```
Comment (TomShawn) [inline br/backup-and-restore-faq.md:63]:
```suggestion
+ **BR 恢复的数据无法被同步到下游**，因为 BR 直接导入 SST 文件，而下游集群目前没有办法获得上游的 SST 文件。
```

File: br/backup-and-restore-faq.md [hunk 4]
```diff
@@ -57,3 +57,11 @@ category: FAQ
...
+
+另一方面，无法被同步到下游的恢复数据可能导致 CDC / Drainer 在执行 DDL 的时候发生异常。所以，如果一定要在 CDC / Drainer 的上游集群执行恢复，请将 BR 恢复的所有表加入 CDC / Drainer 的阻止名单。
+
+CDC 可以通过配置项中的 [`filter.rules`](https://github.com/pingcap/ticdc/blob/7c3c2336f98153326912f3cf6ea2fbb7bcc4a20c/cmd/changefeed.toml#L16) 项完成，Drainer 则可以通过 [`syncer.ignore-table`](https://docs.pingcap.com/zh/tidb/dev/tidb-binlog-configuration-file#ignore-table) 完成。
```
Comment (TomShawn) [inline br/backup-and-restore-faq.md:67]:
```suggestion
TiCDC 可以通过配置项中的 [`filter.rules`](https://github.com/pingcap/ticdc/blob/7c3c2336f98153326912f3cf6ea2fbb7bcc4a20c/cmd/changefeed.toml#L16) 项完成，Drainer 则可以通过 [`syncer.ignore-table`](/tidb-binlog/tidb-binlog-configuration-file.md#ignore-table) 完成。
```

**Comment by TomShawn** [comment]
@YuJuncen Please also involve a technical review.

File: br/backup-and-restore-tool.md [hunk 5]
```diff
@@ -14,6 +14,7 @@ Backup & Restore（以下简称 BR）是 TiDB 分布式备份恢复的命令行
 - BR 只支持 TiDB v3.1 及以上版本。
 - 目前只支持在全新的集群上执行恢复操作。
 - BR 备份最好串行执行，否则不同备份任务之间会相互影响。
+- BR 恢复到 CDC / drainer 的上游集群时，恢复数据无法由 CDC / drainer 同步到下游。
```
Comment (TomShawn) [inline br/backup-and-restore-tool.md:17]:
```suggestion
- BR 恢复到 TiCDC / Drainer 的上游集群时，恢复数据无法由 TiCDC / Drainer 同步到下游。
```

**Comment by YuJuncen** [comment]
@3pointer @leoppro PTAL~

File: br/backup-and-restore-faq.md [hunk 6]
```diff
@@ -57,3 +57,11 @@ category: FAQ
 备份的时候仅仅在每个 Region 的 Leader 处生成该 Region 的备份文件。因此备份的大小等于数据大小，不会有多余的副本数据。所以最终的总大小大约是 TiKV 数据总量除以副本数。
  
 但是假如想要从本地恢复数据，因为每个 TiKV 都必须要能访问到所有备份文件，在最终恢复的时候会有等同于恢复时 TiKV 节点数量的副本。
+
+## BR 恢复到 TiCDC / Drainer 的上游集群时，要注意些什么？
+
++ **BR 恢复的数据无法被同步到下游**，因为 BR 直接导入 SST 文件，而下游集群目前没有办法获得上游的 SST 文件。
+
++ 无法被同步到下游的恢复数据可能导致 CDC / Drainer 在执行 DDL 的时候发生异常。所以，如果一定要在 TiCDC / Drainer 的上游集群执行恢复，请将 BR 恢复的所有表加入 TiCDC / Drainer 的阻止名单。
```
Comment (3pointer) [inline br/backup-and-restore-faq.md:65]: TiCDC 和 前面 CDC 保持统一

**Pull request merged**